task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,187 @@
+      # when KONFLUX-4530 is resolved
+      default: ""
+    - name: PROJECT_NVR
+      description: Name-Version-Release (NVR) of the scanned project, used to find path exclusions
+      type: string
+      default: ""
+    - name: RECORD_EXCLUDED
+      type: string
+      description: |
+        Whether to record the excluded findings (default to false).
+        If `true`, the excluded findings will be stored in `excluded-findings.json`.
+      default: "false"
+    - name: IMP_FINDINGS_ONLY
+      type: string
+      description: Whether to include important findings only
+      default: "true"
+  steps:
+    - name: sast-shell-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.7@sha256:cf6808a3bd605630a5d9f20595ff7c43f8645c00381219d32f5a11e88fe37072
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+      script: |
+        #!/usr/bin/env bash
+        set -x
+        . /utils.sh
+
+        # FIXME: once https://github.com/konflux-ci/konflux-test/pull/303 is available in latest image,
+        # work around below is no longer needed
+        microdnf install -y ShellCheck
+        microdnf download csmock-plugin-shellcheck && rpm -ivh --nodeps ./csmock-plugin-shellcheck-*.rpm && rm -f ./csmock-plugin-shellcheck-*.rpm
+
+        PACKAGE_VERSION=$(rpm -q --queryformat '%{NAME}-%{VERSION}-%{RELEASE}\n' ShellCheck)
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --remove-duplicates
+            --embed-context=3
+            --set-scan-prop="ShellCheck:${PACKAGE_VERSION}"
+        )
+        if [[ "$IMP_FINDINGS_ONLY" == "true" ]]; then
+            # predefined list of shellcheck important findings
+            CSGREP_EVENT_FILTER='\[SC(1020|1035|1054|1066|1068|1073|1080|1083|1099|1113|1115|1127|1128|1143|2043|2050|'
+            CSGREP_EVENT_FILTER+='2055|2057|2066|2069|2071|2077|2078|2091|2092|2157|2171|2193|2194|2195|2215|2216|'
+            CSGREP_EVENT_FILTER+='2218|2224|2225|2242|2256|2258|2261)\]$'
+            CSGREP_OPTS+=(
+                --event="$CSGREP_EVENT_FILTER"
+            )
+        else
+            CSGREP_OPTS+=(
+                --event="info|warning"
```

Comment:
This should have been `error|warning`.  We definitely do **not** want to suppress the findings with `error` events, which denote the highest severity.